hooks/useParseBigNumber.ts

```diff
@@ -0,0 +1,18 @@
+import { BigNumber } from '@ethersproject/bignumber'
+import { BigNumber as BN } from 'bignumber.js'
+import { useMemo } from 'react'
+
+export default function useParseBigNumber(
+  value: string | undefined,
+  decimals = 0,
+): BigNumber {
+  return useMemo(() => {
+    if (!value) return BigNumber.from(0)
+    try {
+      return BigNumber.from(new BN(value).shiftedBy(decimals).toFixed(0))
```

Comment:
This function should consider negative BigNumber as incorrect and return zero:
<img width="451" alt="Screenshot 2022-12-20 at 16 04 58" src="https://user-images.githubusercontent.com/5823445/208627286-96dda8a1-0b2b-45eb-9a67-b801e5564c0a.png">

```suggestion
      const valueBN = BigNumber.from(new BN(value).shiftedBy(decimals).toFixed(0))
      if (valueBN.lt(0)) {
        console.error('Value is negative')
        return BigNumber.from(0)
      }
      return valueBN
```